### PR TITLE
MAINT make pyarrow optional

### PR DESCRIPTION
**Pull request by glemaitre**
Make `pyarrow` optional such that it can be imported with `pyodide`.

### CODE REVIEW

File: skrub/datasets/_fetching.py [hunk 1]
```diff
@@ -371,6 +371,10 @@ def _fetch_figshare(
             UserWarning,
             stacklevel=2,
         )
+        import_optional_dependency(
+            "pyarrow", extra="pyarrow is required for parquet support."
+        )
```
Comment (GaelVaroquaux) [inline skrub/datasets/_fetching.py:376]: I think that I'd prefer to implement the try-except ourselves rather than import from a private pandas module.

**Review by GaelVaroquaux — CHANGES_REQUESTED** [review]
One minor change requested, plus:

- add in the docstring of all the functions skrub.datasets.fetch_ken* that they require the optional dependency pyarrow
- Add a note saying the same thing on top of https://skrub-data.org/stable/auto_examples/06_ken_embeddings_example.html

**Comment by glemaitre** [comment]
I added in the "Notes" section that `pyarrow` is required for some of the functions.
I also added a note in the example as suggested.

**Comment by GaelVaroquaux** [comment]
Can you add a changelog

**Pull request merged**